libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java

```diff
@@ -105,61 +173,15 @@ public Future<List<ClassifiedSegment>> loadSegments(final List<String> userIds,
      * @param segments classified segments of a group of users from the database.
      * @return the classified segments as POJO list.
      */
-    private List<ClassifiedSegment> pojo(final List<JsonObject> segments) {
+    private List<? extends ClassifiedSegment> pojo(final List<JsonObject> segments) {
         if (segments.size() == 0) {
             return new ArrayList<>();
         }
 
         // Convert documents
-        final var pojoSegments = segments.stream().map(this::toSegment);
-        return pojoSegments.collect(Collectors.toList());
-    }
-
-    /**
-     * Constructs a fully initialized {@link ClassifiedSegment} from a database entry.
-     *
-     * @param segment The entry from the database.
-     * @return the created {@code ClassifiedSegment}
-     */
-    private ClassifiedSegment toSegment(final JsonObject segment) {
-        final var oid = segment.getJsonObject("_id").getString("$oid");
-        final var forward = segment.getBoolean("forward");
-        final var geometry = segment.getJsonObject("geometry");
-        final var length = segment.getDouble("length");
-        final var modality = Modality.valueOf(segment.getString("modality"));
-        final var way = segment.getLong("way");
-        final var vnk = segment.getLong("vnk");
-        final var nnk = segment.getLong("nnk");
-        final var wayOffset = segment.getDouble("way_offset");
-        final var tags = segment.getJsonObject("tags").getMap();
-        final var latestDataPoint = OffsetDateTime.parse(segment.getJsonObject("latest_data_point").getString("$date"));
-        final var userId = segment.getString("userId");
-        final var expectedValue = segment.getDouble("expected_value");
-        final var variance = segment.getDouble("variance");
-        // We're expecting a large number of segments stored, we store the quality class as Integer instead of String.
-        // This should reduce the collection size (at least when uncompressed).
-        // Later on we might store this as a normalized value between zero and one.
-        final var quality = ClassifiedSegment.SurfaceQuality.valueOf(segment.getInteger("quality"));
-        final var dataPointCount = segment.getLong("data_point_count");
-        final var version = segment.getString("version");
-        return new ClassifiedSegment(oid, forward, toGeometry(geometry), length, modality, vnk, nnk, wayOffset, way,
-                tags, latestDataPoint, userId, expectedValue, variance, quality, dataPointCount, version);
-    }
+        final var factory = ClassifiedSegmentFactoryProvider.getFactory(mode);
```

Comment:
this is why I need to inject a "mode" into the constructor of this class, to choose a factory here.

Reply:
Why can't the mode just be the correct factory implementation?

Reply:
True. I changed this.